src/argilla/client/feedback/training/schemas.py

```diff
@@ -982,6 +1015,92 @@ def _prepare_for_training_with_trl(
         return ds
 
 
+class TrainingTaskForQuestionAnsweringFormat(BaseModel):
+    """
+    Union[Tuple[str, str, str], List[Tuple[str, str, str]]]
+    """
+
+    format: Union[Tuple[str, str, str], List[Tuple[str, str, str]]]
+
+
+class TrainingTaskForQuestionAnswering(BaseModel, TrainingData):
+    _formatting_func_return_types = TrainingTaskForQuestionAnsweringFormat
+    formatting_func: Optional[Callable[[Dict[str, Any]], Union[None, str, Iterator[str]]]] = None
+    question: Optional[TextField] = None
+    context: Optional[TextField] = None
+    answer: Optional[TextQuestion] = None
+
+    def _format_data(self, dataset: "FeedbackDataset") -> List[Dict[str, str]]:
+        if self.formatting_func is not None:
+            output = set()
+            for sample in dataset.format_as("datasets"):
+                question_context_answer = self.formatting_func(sample)
+                if question_context_answer is None:
+                    continue
+
+                self._test_output_formatting_func(question_context_answer)
+
+                if isinstance(question_context_answer, tuple):
+                    question_context_answer = {question_context_answer}
+
+                output |= set(question_context_answer)
+            return [
+                {"question": question, "context": context, "answer": answer} for question, context, answer in output
+            ]
+        else:
+            return super()._format_data(dataset)
+
+    @property
+    def supported_frameworks(self) -> List[Framework]:
+        names = ["transformers"]
+        return [Framework(name) for name in names]
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}\n\t formatting_func={self.formatting_func}"
+
+    @requires_version("transformers")
+    def _prepare_for_training_with_transformers(
+        self, data: List[dict], train_size: float, seed: int, framework=None
+    ) -> Union["datasets.Dataset", "datasets.DatasetDict"]:
+        import datasets
+
+        datasets_dict = {
+            "question": [],
+            "context": [],
+            "answer": [],
+        }
+        for entry in data:
+            if any([entry["question"] is None, entry["context"] is None, entry["answer"] is None]):
+                continue
+            if entry["answer"] not in entry["context"]:
+                continue
+            # get index of answer in context
+            answer_start = entry["context"].index(entry["answer"])
```

Comment:
Is the answer included in the context?

Reply:
Yes, because it is extractive QnA, it is supposed to be there. Maybe a warning would be better?

Reply:
Ooh, that makes sense